service/src/services/LibraryService.ts

```diff
@@ -124,15 +131,40 @@ export class LibraryService implements Service<fhir4.Library> {
     const query = extractIdentificationForQuery(args, params);
     const parsedParams = parseRequestSchema({ ...params, ...query }, LibraryDataRequirementsArgs);
 
+    // check to see if data requirements were already calculated for this Library and params
+    const dataReqsQuery: Filter<any> = {};
+    Object.entries(parsedParams).forEach(p => {
+      if (p[0] === 'id') {
+        dataReqsQuery.resourceId = p[1] as string;
+      } else {
+        dataReqsQuery[p[0]] = p[1] as string;
+      }
+    });
+
+    const dataReqs = await findDataRequirementsWithQuery(dataReqsQuery);
+
+    // if data requirements were already calculated for this Library and params, return them
+    if (dataReqs) {
+      logger.info('Successfully retrieved $data-requirements report');
+      return dataReqs;
+    }
+
```

Comment:
As we build out the data requirements functionality between the Library and Measure service, it looks like we are reusing a lot of code. Do you think it's possible to pull some of this out into a function that can be used by both services?